File: examples/mpi/hellompi.c
#include <mpi.h>
#include <stdio.h>
#include <string.h>

int main(int argc, char** argv)
{
  int rank, size, i, tag;
  MPI_Status status;
  char message[20];

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &size);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

  tag = 100;

  if (rank == 0) {
    strcpy(message, "Hello world!");
    for (i=1; i < size; ++i)
      MPI_Send(message, 13, MPI_CHAR, i, tag, MPI_COMM_WORLD);
  } else
    MPI_Recv(message, 13, MPI_CHAR, 0, tag, MPI_COMM_WORLD, &status);

  printf("process %d: %s\n", rank, message);

  MPI_Finalize();
  return 0;
}

File: examples/mpi/hellompi.f90
program hello
  include 'mpif.h'
  integer rank, size, ierror, tag
  integer status(MPI_STATUS_SIZE)
  character(12) message
  call MPI_INIT (ierror)
  call MPI_COMM_SIZE (MPI_COMM_WORLD, size, ierror)
  call MPI_COMM_RANK (MPI_COMM_WORLD, rank, ierror)
  tag = 100
  if (rank .eq. 0) then
    message = 'Hello world!'
    do i=1, size-1
      call MPI_SEND (message, 12, MPI_CHARACTER, i, tag, MPI_COMM_WORLD, ierror)
  enddo
  else
    call MPI_RECV (message, 12, MPI_CHARACTER, 0, tag, MPI_COMM_WORLD, status, ierror)
  endif
  print *, 'process', rank, ': ', message
  call MPI_Finalize (ierror)
end
